### PR TITLE
Mailroom_pt3 submission

### PR DESCRIPTION
**Pull request by tfbanks**
submitting mailroom_pt3

### CODE REVIEW

File: students/tfbanks/Lesson05/Mailroom_pt3.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            print('''Enter the actions's associated number''')
+        if answer == 1:
+            thank_you()
+        elif answer == 2:
+            report()
+        elif answer == 3:
+            mass_mail()
+        elif answer == 4:
+            print('Have a nice day')
+            break
```
Comment (natasha-aleksandrova) [inline students/tfbanks/Lesson05/Mailroom_pt3.py]: Why was this changed back to previous versions instead of building on top of it? This used to be a switch dict style in a previous submission. I would recommend changing it back.

File: students/tfbanks/Lesson05/Mailroom_pt3.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    else:
+        donor_names[ty_donor] = [amount]
+
+    print(letter.format(ty_donor, amount))
+
+
+def report():  # Defines parameters of the report of donors, creates a new list of summarized details per donor, sorts, and prints it.
+    print('\nDonor Name            |  Total Given | Num Gifts |  Average Gift')
+    print('------------------------------------------------------------------')
+    donor_rpt = [[donor, sum(donations), len(donations), sum(donations)/len(donations)] for donor, donations in donor_names.items()]
```
Comment (natasha-aleksandrova) [inline students/tfbanks/Lesson05/Mailroom_pt3.py]: Usually you want to avoid one letter vars, or short vars but in the case of comprehensions it's ok.

Very common pattern is to use `for k, v in some_dict.items()` or `for key, val in some_dict.items()`
so that the comprehension line is not overly long.

File: students/tfbanks/Lesson05/Mailroom_pt3.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        print('\n')
+        ty_donor = input('Enter the Full Name of a donor (for list of previous donors type list): ').title()
+
+    while True:
+        try:
+            amount = float(input('Please enter the donation amount: '))
+        except ValueError:
+            print('Please enter a valid donation amount')
+            continue
+        break
```
Comment (natasha-aleksandrova) [inline students/tfbanks/Lesson05/Mailroom_pt3.py]: you can remove the `continue` and add `break` into `else` line like so:
```
try:
    # try something
except ValueError:
    print('error')
else:
    break
```

**Comment by tfbanks** [comment]
In one of the comments made with an earlier submittal made me rethink how that switch dict code was working and playing around with it with the try except and removing all the call backs at the end of each selection I thought this was a good option, i will switch it back with the lesson six exercise.

**Comment by natasha-aleksandrova** [comment]
I reviewed the comments and not sure which one that was, apologies if there was confusion.

Let me know if there are other question on that. Switch dict pattern is still a good option here so I would stick with it (reduces boiler plate code).

**Comment by tfbanks** [comment]
I was looking at the comment about “might as well call the function here too…” 

I saw the select (answer) and then the comment after your suggested while true statement, and I thought you were referring to it being ok to use select(answer) and then thought about  just calling the functions there by their selection is what you were referring to because that also made sense to me.

 

That’s ok though,  learned a lot in the process
